CMakeLists.txt

```diff
@@ -1,18 +1,23 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(eigen_stl_containers)
 
-find_package(catkin REQUIRED)
+if(NOT WIN32)
+  add_definitions(-std=c++11)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-# We don't build anything here, but in order to export the right
-# build flags in catkin_package(), we still have to find Eigen3 here.  Note
-# that this is somewhat complicated because of our need to support Ubuntu
-# all the way back to saucy.  First we look for the Eigen3 cmake module
-# provided by the libeigen3-dev on newer Ubuntu.  If that fails, then we
-# fall-back to the version provided by cmake_modules, which is a stand-in.
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
 find_package(Eigen3 QUIET)
```

Comment:
We can just make this:

```
find_package(Eigen3 REQUIRED)
```

And get rid of all of the backwards-compatibility stuff; this will never go onto older Ubuntu versions.